grpc/dispatcher.go

```diff
@@ -0,0 +1,86 @@
+package grpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/gojek/fiber"
+	fiberError "github.com/gojek/fiber/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	TimeoutDefault = time.Second
+)
+
+type Dispatcher struct {
+	Timeout time.Duration
+}
+
+func (d *Dispatcher) Do(request fiber.Request) fiber.Response {
+
+	grpcRequest, ok := request.(*Request)
+	if !ok {
+		return fiber.NewErrorResponse(
+			fiberError.FiberError{
+				Code:    int(codes.InvalidArgument),
+				Message: "fiber: grpc.Dispatcher supports only grpc.Request type of requests",
+			})
+	}
+
+	if grpcRequest.hostport == "" || grpcRequest.ServiceMethod == "" {
```

Comment:
Will `hostport` and `ServiceMethod` be different for each call? Could it be part of the grpc fiber config?

Reply:
It can be. It depends how "generic" we want a dispatcher to be. 

As per above, i think we will put `hostport` with the dispatcher to prevent multiple connection creation.

The benefit of putting these in the request, one single dispatcher can be used for multiple calls. But thinking about it now, we create one dispatcher per route (one endpoint and method). So there isn't much point granting these flexibility and can be park with the dispatcher for now.

Will refactor these

Reply:
moved